compiler/rustc_hir/src/hir.rs

```diff
@@ -1258,7 +1258,7 @@ pub struct Arm<'hir> {
     /// If this pattern and the optional guard matches, then `body` is evaluated.
     pub pat: &'hir Pat<'hir>,
     /// Optional guard clause.
-    pub guard: Option<Guard<'hir>>,
```

Comment:
Ah, I've always been so confused why this was represented differently in HIR. This was also what caused the bug fixed by #119402, for example.

Reply:
I think if_let_guards was implemented before let chains were ready. But it's definitely outstayed its usefulness.